plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/remote/add/AddRemoteRepositoryPresenter.java

```diff
@@ -28,21 +30,38 @@
  */
 @Singleton
 public class AddRemoteRepositoryPresenter implements AddRemoteRepositoryView.ActionDelegate {
-    private AddRemoteRepositoryView view;
-    private GitServiceClient        service;
-    private AppContext              appContext;
-    private AsyncCallback<Void>     callback;
 
-    /**
-     * Create presenter.
-     *
-     * @param view
-     * @param service
-     * @param appContext
-     */
+    // An alternative scp-like syntax: [user@]host.xz:path/to/repo.git/
+    private static final RegExp SCP_LIKE_SYNTAX        = RegExp.compile("([A-Za-z0-9_\\-]+\\.[A-Za-z0-9_\\-:]+)+:");
+    // the transport protocol
+    private static final RegExp PROTOCOL               = RegExp.compile("((http|https|git|ssh|ftp|ftps)://)");
+    // the address of the remote server between // and /
+    private static final RegExp HOST1                  = RegExp.compile("//([A-Za-z0-9_\\-]+\\.[A-Za-z0-9_\\-:]+)+/");
+    // the address of the remote server between @ and : or /
+    private static final RegExp HOST2                  = RegExp.compile("@([A-Za-z0-9_\\-]+\\.[A-Za-z0-9_\\-:]+)+[:/]");
+    // the repository name
+    private static final RegExp REPO_NAME              = RegExp.compile("/[A-Za-z0-9_.\\-]+$");
+    // start with white space
+    private static final RegExp START_WITH_WHITE_SPACE = RegExp.compile("^\\s");
+    // remote name
+    private static final RegExp REMOTE_NAME            = RegExp.compile("^[A-Za-z0-9_\\.]+$");
```

Comment:
why dash is not included?
`git remote add gaal-1 git@github.com:garagatyi/che-parent.git` works in terminal